rfcs/react-components/components/combobox-spacebar.md

```diff
@@ -0,0 +1,115 @@
+
+The content of options also influences user expectations -- there would be a greater expectation of space-to-type when options contain a space in them, for example multi-word options, names, states or cities.
+
+One of the risks is that the developers and designers may not be in a position to make a correct decision on the value of the prop. This is because options are usually either translations or user data, which may contain spaces that the authoring team was not aware of. We generally can't expect authors to understand all possible translations across supported languages or the specific validations of the user data at UI authoring time. While the occaisonal space character does not necessarily mean that space should not select, translations and user data do introduce the possibility of frequent spaces within multiple options.
+
+## Problem statement
+
+We cannot fully know whether the ideal spacebar behavior is to select or insert a character, since we don't know the full context of usage. Our choices mostly revolve around balancing providing more flexibility vs. more consistency, and choosing the best default if we do provide flexibility.
+
+## Options
+
+### 1. `allowSpaceCharacter` prop, default to `true`
+
+This is the option that allows authors to choose the best behavior for their use case, defaulting to allowing space to insert a character. This borrows the behavior from the freeform PR, where space will still select if the user is actively interacting with the dropdown listbox (i.e. it is open, and the last interaction was to navigate through options).
+
+We would likely include author guidance that all Comboboxes within their app use the same behavior for `allowSpaceCharacter`.
+
+#### Pros
+
+- Allows authors to choose the best behavior per use case
+- Defaulting to `true` means the default probably(?) matches the majority of Combobox usage
+- Defaulting to `true` means that often-missed use cases like spaces in translated strings or user-generated options won't cause issues
+
+#### Cons
+
+- Defaulting to `true` means the simplest/most basic use case is not the default
+- Defaulting to `true` means the Combobox space behavior is different from Dropdown by default, which primarily affects keyboard and screen reader users.
+- Providing a prop to alter the space key behavior potentially means inconsistent behavior within an app, or across related apps
+- This behavior is not aligned with the results of the user study
+
+### 2. `allowSpaceCharacter` prop, default to `false`
+
+This is the same as (1), but with a different default. The Pros/Cons list is essentially the same, but reversed.
```

Comment:
I don't think having a prop to configure this behavior is the right first step. I think we need to get the default behavior correct, and then if there's a compelling reason to add a prop to differ from the default behavior, then we can add it later. But relying on devs to pick the right behavior on this nuanced issue, on every combobox in every app doesn't seem like it'll work.